reactive/easyrsa.py

```diff
@@ -212,21 +214,56 @@ def send_ca(tls):
     tls.set_client_cert(client_cert, client_key)
 
 
+@when('leadership.is_leader',
+      'easyrsa.certificate.authority.available',
+      'client.available')
+@when_not('easyrsa.global-client-cert.created')
+def publish_global_client_cert():
+    """
+    This is for backwards compatibility with older tls-certificate clients
+    only.  Obviously, it's not good security / design to have clients sharing
+    a certificate, but it seems that there are clients that depend on this
+    (though some, like etcd, only block on the flag that it triggers but don't
+    actually use the cert), so we have to set it for now.
+    """
+    tls = endpoint_from_flag('client.available')
+    client_cert, client_key = create_client_certificate()
+    tls.set_client_cert(client_cert, client_key)
+    set_flag('easyrsa.global-client-cert.created')
+
+
 @when('client.server.cert.requested', 'easyrsa.configured')
-def create_server_cert(tls):
+def create_server_cert():
     '''Create server certificates with the request information from the
     relation object.'''
-    # Get the map of unit names to requests.
-    requests = tls.get_server_requests()
-    # Iterate over all items in the map.
-    for unit_name, request in requests.items():
-        cn = request.get('common_name')
-        sans = request.get('sans')
-        name = request.get('certificate_name')
+
+    tls = endpoint_from_flag('client.server.cert.requested')
+
+    # Iterate over all new requests
+    for request in tls.new_server_requests:
+        cn = request.common_name
+        sans = request.sans
+        name = request.common_name
         # Create the server certificate based on the information in request.
         server_cert, server_key = create_server_certificate(cn, sans, name)
         # Set the certificate and key for the unit on the relationship object.
-        tls.set_server_cert(unit_name, server_cert, server_key)
+        request.set_cert(server_cert, server_key)
+
+
+@when('client.client.certs.requested', 'easyrsa.configured')
+def create_client_cert():
+    '''Create client certificates with the reuqest information from the
+    relation object.'''
+
+    tls = endpoint_from_flag('client.client.cert.requested')
+
+    # Iterate over all new requests
+    for request in tls.new_client_requests:
+        # Create a client certificate for this request.
+        name = request.get('certificate_name')
```

Comment:
This is unlikely to be set, meaning all client certs will [end up using the same cert](https://github.com/juju-solutions/layer-easyrsa/blob/master/reactive/easyrsa.py#L305).  That linked check also seems like it will never regenerate certs if the SANs change.  I think the easy fix is to just generate a nonce in place of the name and then ensure we delete the generated certs after they're read.